helix-term/src/ui/editor.rs

```diff
@@ -152,6 +153,37 @@ impl EditorView {
         self.render_statusline(doc, view, statusline_area, surface, theme, is_focused);
     }
 
+    pub fn render_rulers(
+        editor: &Editor,
+        doc: &Document,
+        view: &View,
+        viewport: Rect,
+        surface: &mut Surface,
+        theme: &Theme,
+    ) {
+        let editor_rulers = &editor.config().rulers;
+
+        let rulers = doc
+            .language_config()
+            .and_then(|config| config.rulers.as_ref())
+            .unwrap_or(editor_rulers);
+
+        rulers
+            .iter()
+            .filter_map(|ruler| {
+                ruler
+                    .checked_sub(1 + view.offset.col as u16)
+                    .map(|x| {
+                        viewport
+                            .with_height((view.last_line(doc) - view.offset.row + 1) as u16)
+                            .clip_left(x)
+                            .with_width(1)
+                    })
+                    .filter(|area| area.left() < viewport.right())
+            })
+            .for_each(|area| surface.set_style(area, theme.get("ui.highlight.ruler")));
```

Comment:
```suggestion
        rulers
            .iter()
            // View might be horizontally scrolled, convert from absolute distance
            // from the 1st column to relative distance from left of viewport
            .filter_map(|ruler| ruler.checked_sub(1 + view.offset.col as u16))
            .filter(|ruler| ruler < &viewport.width)
            .map(|ruler| viewport.clip_left(ruler).with_width(1))
            .for_each(|area| surface.set_style(area, theme.get("ui.highlight.ruler")))
```

Reply:
I'd also pull out the theme.get call so we're not re-fetching on every iteration

Reply:
Thank you! This looks a lot nicer! There is just one thing I am wondering about. You dropped the `with_height`. Is that on purpose? Because it draws the ruler on the whole screen instead of stopping at the last line of text

![height-ruler](https://user-images.githubusercontent.com/74479846/163724100-dd24ace2-abde-4785-af2e-c546869693b0.png)

Is that the wanted behaviour?

Isn't it a bit nicer with `.map(|ruler| viewport.clip_left(ruler).with_width(1).with_height((view.last_line(doc) - view.offset.row + 1) as u16)`?

![height-ruler](https://user-images.githubusercontent.com/74479846/163724292-e7601794-400c-4890-995e-a64871e1d7e4.png)

Reply:
I personally like the version which extends to the whole screen since it feels consistent (not a strong argument for a vertical colored line I know :)